src/background/services/firebase/FirebaseService.ts

```diff
@@ -98,6 +112,58 @@ export class FirebaseService {
     this.#fcmMessageEventEmitter.on(event, listener);
   }
 
+  #getModel({ tools, toolConfig, systemInstruction }: ConfigParams) {
+    if (!this.#app) {
+      throw new Error('Firebase app has not initialized');
+    }
+    const vertexAI = getVertexAI(this.#app);
+    const generationConfig = {
+      topP: 0.95,
+      topK: 40,
+      maxOutputTokens: 8192,
+    };
+    this.#config = {
+      generationConfig,
+      tools,
+      toolConfig,
+      systemInstruction,
+    };
+
+    return getGenerativeModel(vertexAI, {
+      model: 'gemini-2.0-flash',
+      generationConfig,
+      tools,
+      toolConfig,
+      systemInstruction,
+    });
+  }
+
+  async generateContent(message: string, parts?: Content[]) {
+    const contents = parts
+      ? ([{ role: 'user', parts: [{ text: message }] }, ...parts] as Content[])
+      : ([{ role: 'user', parts: [{ text: message }] }] as Content[]);
+    const result = await this.#model?.generateContent({
+      ...this.#config,
+      contents,
+    });
+    if (!result) {
+      throw new Error('no result');
+    }
+    const response = result.response;
+    return response;
+  }
+
+  async startChat({ tools, toolConfig, systemInstruction }: ConfigParams) {
+    if (!this.#model) {
+      this.#model = this.#getModel({
+        tools,
+        toolConfig,
+        systemInstruction,
+      });
+    }
```

Comment:
With this check, it's impossible to update the `systemInstruction`, so when the user switches network, account or the balances are updated, the whole model becomes outdated and will likely cause issues.
We need to be able to update the `systemInstruction` here.